iceoryx_hoofs/include/iceoryx_hoofs/cxx/functional_interface.hpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_CXX_FUNCTIONAL_POLICY_HPP
+#define IOX_HOOFS_CXX_FUNCTIONAL_POLICY_HPP
+
+#include "iceoryx_hoofs/cxx/function_ref.hpp"
+#include "iceoryx_hoofs/cxx/helplets.hpp"
+#include <iostream>
+
+namespace iox
+{
+namespace cxx
+{
+namespace internal
+{
+template <typename T, class = void>
+struct HasValueMethod : std::false_type
+{
+};
+
+template <typename T>
+struct HasValueMethod<T, cxx::void_t<decltype(std::declval<T>().value())>> : std::true_type
+{
+};
+
+template <typename T, class = void>
+struct HasGetErrorMethod : std::false_type
+{
+};
+
+template <typename T>
+struct HasGetErrorMethod<T, cxx::void_t<decltype(std::declval<T>().get_error())>> : std::true_type
+{
+};
+
+template <typename T>
+struct Expect
+{
+    /// @brief Expects that the object is valid, otherwise the method prints the
+    ///        provided message and induces a fatal error
+    /// @param[in] msg Message which will be print when the object is invalid
```

Comment:
```suggestion
    /// @param[in] msg Message which will be printed when the object is invalid
```
Please correct this also in all comments in the `ExpectWithValue` struct.